src/middleware/authentication.middleware.ts

```diff
@@ -39,18 +40,27 @@ export const authentication = (req: Request, res: Response, next: NextFunction):
   }
 };
 
-function getReturnToUrl(path: string) {
+function getReturnToUrl(req: Request) {
   let returnToUrl = SOLD_LAND_FILTER_URL;
+  const path = req.path;
 
   if (path === STARTING_NEW_URL || path.endsWith(`/${RESUME}`) || path === UPDATE_CONTINUE_WITH_SAVED_FILING_URL) {
     if (!path.startsWith(REGISTER_AN_OVERSEAS_ENTITY_URL) && !path.startsWith(UPDATE_AN_OVERSEAS_ENTITY_URL)) {
       throw new Error('Security failure with the path URL ' + path);
     }
 
     returnToUrl = path;
+
+    if (req.query[JOURNEY_QUERY_PARAM] === JourneyType.remove) {
+      // Ensure that user is placed on the Remove (not Update) journey after logging in and note that URL now needs to
+      // be encoded due to the presence of the additional '?', required for the 'remove journey' query parameter
+      returnToUrl = encodeURIComponent(returnToUrl + `?${JOURNEY_QUERY_PARAM}=${JourneyType.remove}`);
+    }
   } else if (path.startsWith(UPDATE_LANDING_URL)) {
     returnToUrl = SECURE_UPDATE_FILTER_URL;
```

Comment:
does the else block need changing also on line 60, I think some (all?) remove screens will start with  UPDATE_LANDING_URL

Reply:
I think we agreed to leave this where it is (at least for now).